### DISCUSSION TITLE
How do I use the Access variable?

### DISCUSSION THREAD

**Discussion by laurentmc0906** [nosoop/SM-TFCustomWeaponsX#64] [Q&A] [answered]
I'm not that good with servers and I'm a little confused..

**Comment by nosoop** [comment, marked as answer]
It's fairly difficult to explain unless you have a general understanding of the more granular admin system, and the entire process varies depending on what you're using for admin configuration (here I'll link to specifics using the flat file system, but SourceMod does support admin configuration via database as well, not to mention the available community admin tools).

Note that you don't have to worry about access restrictions unless you need to restrict availability the item to specific users &mdash; if omitted everyone can access them by default.

The admin system has concepts of users, groups, and command strings.  You create a group, assign command string privileges to said group, then assign groups to users.

In short:

1. You think up of an arbitrary value like `admin_cwx_weapon`, set the `access` value in the weapon configuration to that.  You can reuse that same value across items.
2. In the admin group configuration, you create or use an existing admin group, then assign a command string privilege to allow `admin_cwx_weapon`.  [You can refer to this page if you are using the flat file method.](https://wiki.alliedmods.net/Adding_Groups_(SourceMod)#File_Format)
3. In the admin config, you create an admin entry and bind it to a specific user, then you add the admin group from the previous step to that entry.  [See this page for the flat file method.](https://wiki.alliedmods.net/Adding_Admins_(SourceMod)#Detailed_Admins)

**Reply by laurentmc0906** [reply]
Thank you so much! I do have another question, unrelated to this plugin, but is still your plugin.

**Reply by laurentmc0906** [reply]
> Thank you so much! I do have another question, unrelated to this plugin, but is still your plugin.

https://github.com/nosoop/SM-TFAttributeSupport Does the override projectile still work? I'm trying to change a rocket launcher to have A grenade projectile, and the velocity is fucked on it.. But this plugin should fix that, right?,, or am I misinterpreting it..?

**Reply by nosoop** [reply]
That's something that would be better served with an issue in that repository, but yes, the plugin should be handling that.

Note that the plugin was failing to load on Linux unless you applied the fixes in nosoop/SM-TFAttributeSupport#28.  Should be fixed shortly.
